maskrcnn_benchmark/structures/bounding_box.py

```diff
@@ -213,10 +225,12 @@ def __len__(self):
 
     def clip_to_image(self, remove_empty=True):
         TO_REMOVE = 1
-        self.bbox[:, 0].clamp_(min=0, max=self.size[0] - TO_REMOVE)
-        self.bbox[:, 1].clamp_(min=0, max=self.size[1] - TO_REMOVE)
-        self.bbox[:, 2].clamp_(min=0, max=self.size[0] - TO_REMOVE)
-        self.bbox[:, 3].clamp_(min=0, max=self.size[1] - TO_REMOVE)
+        self.bbox = torch.stack([
+            self.bbox[:, 0].clamp(min=0, max=self.size[0] - TO_REMOVE),
+            self.bbox[:, 1].clamp(min=0, max=self.size[1] - TO_REMOVE),
+            self.bbox[:, 2].clamp(min=0, max=self.size[0] - TO_REMOVE),
+            self.bbox[:, 3].clamp(min=0, max=self.size[1] - TO_REMOVE),
+        ], dim=1)
```

Comment:
This could have a performance impact (we replace inplace by non-inplace+stack). Should I make it conditional on tracing?

Reply:
I'm not sure this is worth the trouble, the different might not be significant.
And actually, now that we are at it, we could even (maybe) optimize it slightly by doing something like
```python
x1 = self.bbox[:, 0::2].clamp(min=0, max=self.size[0] - TO_REMOVE)
x2 = self.bbox[:, 1::2].clamp(min=0, max=self.size[1] - TO_REMOVE)
res = torch.stack([x1, x2], dim=2).view(...)
```

but I'm not sure it's worth it.

Reply:
I like it, but I'll call it `xs` and `ys`.